src/lib/mdns/ServicePool.h

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/mdns/platform/Mdns.h>
+
+namespace chip {
+namespace Mdns {
+
+constexpr uint64_t kUndefinedNodeId = 0;
+
+class ServicePool
```

Comment:
This needs documentation on proper use.  In particular, the following should definitely be documented:

1. This class does not do any automatic cleanup.  If you create it, add some things, then just let it go out of scope without calling `Clear()`, it will leak.  Should `DiscoveryManager` be calling `Clear()` somewhere?
2. Something about the ownership of the thing returned from `FindService`.  In particular that it's returning a pointer to something owned by this class, and that consumers who want to keep it across any further API calls to this class should make a copy of whatever data they need.